src/property.hpp

```diff
@@ -139,10 +140,8 @@ namespace realm {
                 return "array";
             case PropertyType::LinkingObjects:
                 return "linking objects";
-#if __GNUC__
```

Comment:
Um, I'm almost positive Clang also defines `__GNUC__` - or at least Microsoft's Clang/C2 does. I can revert this change if it only makes sense for GCC, though.